codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/Instantiator.kt

```diff
@@ -122,28 +123,50 @@ open class Instantiator(
         fun doesSetterTakeInOption(memberShape: MemberShape): Boolean
     }
 
+    /**
+     * Customize how each shape is rendered, instead of relying on static `Node` data
+     */
+    open class CustomWritable {
+        // Return `null` for the default behavior
+        open fun render(shape: Shape): Writable? = null
```

Comment:
We use `render` when we actually render code, so, when we have a `RustWriter` and we write it. We use `generate` or other words when we return a `Writable` that gets composed and rendered afterwards.